hail/python/hailtop/auth/auth.py

```diff
@@ -128,34 +136,59 @@ def create_user(username: str, login_id: str, is_developer: bool, is_service_acc
+async def async_create_user(username: str, login_id: str, is_developer: bool, is_service_account: bool, namespace: Optional[str] = None, *, wait: bool = False):
     deploy_config, headers, _ = deploy_config_and_headers_from_namespace(namespace)
 
-    body = {
-        'login_id': login_id,
-        'is_developer': is_developer,
-        'is_service_account': is_service_account,
-    }
+    try:
+        body = {
+            'login_id': login_id,
+            'is_developer': is_developer,
+            'is_service_account': is_service_account,
+        }
 
-    async with aiohttp.ClientSession(
-            raise_for_status=True,
-            timeout=aiohttp.ClientTimeout(total=30),
-            headers=headers) as session:
-        await request_retry_transient_errors(
-            session, 'POST', deploy_config.url('auth', f'/api/v1alpha/users/{username}/create'), json=body
-        )
+        async with aiohttp.ClientSession(
+                raise_for_status=True,
+                timeout=aiohttp.ClientTimeout(total=30),
+                headers=headers) as session:
+            await request_retry_transient_errors(
+                session, 'POST', deploy_config.url('auth', f'/api/v1alpha/users/{username}/create'), json=body
+            )
+
+        if wait:
+            await _wait_user_state_transition(username, namespace, 'creating', 'active')
+            print(f'Created user {username}')
+    except Exception as e:
+        raise CreateUserException(f"Error while creating user '{username}'") from e
 
 
 def delete_user(username: str, namespace: Optional[str] = None):
     return async_to_blocking(async_delete_user(username, namespace=namespace))
 
 
-async def async_delete_user(username: str, namespace: Optional[str] = None):
+async def async_delete_user(username: str, namespace: Optional[str] = None, *, wait: bool = False):
     deploy_config, headers, _ = deploy_config_and_headers_from_namespace(namespace)
-    async with aiohttp.ClientSession(
-            raise_for_status=True,
-            timeout=aiohttp.ClientTimeout(total=300),
-            headers=headers) as session:
-        await request_retry_transient_errors(
-            session, 'DELETE', deploy_config.url('auth', f'/api/v1alpha/users/{username}')
-        )
+
+    try:
+        async with aiohttp.ClientSession(
+                raise_for_status=True,
+                timeout=aiohttp.ClientTimeout(total=300),
+                headers=headers) as session:
+            await request_retry_transient_errors(
+                session, 'DELETE', deploy_config.url('auth', f'/api/v1alpha/users/{username}')
+            )
+
+        if wait:
+            await _wait_user_state_transition(username, namespace, 'deleting', 'deleted')
+            print(f'Deleted user {username}')
+    except Exception as e:
+        raise DeleteUserException(f"Error while deleting user '{username}'") from e
```

Comment:
Most of this stuff is just refactoring. Pushing what was in the `{create|delete}_user.py` files into here so that I could have `wait=True` from python code and not just the CLI